### PR TITLE
Add R Lintr

### PR DESCRIPTION
**Pull request by emichaud998**
Changes in this PR:
This PR adds a lintr for the R code in the shinydashboard/lantern directory. This lintr runs within a script that exits with a failed status code of 1 if the lintr command returns with any lintr problems. This PR also cleaned up some of the R code according to the lintr tests, and it adds the R lintr to travis. In the make file, the make lint command now includes the R lintr, and now there are two separate make commands, make lint_go and make lint_R, which run the linters for their corresponding language. 

This PR also has updated travis to run all the tests (make test, make test_int, and make test_e2e) in one stage without needing to rebuild the docker image a second time to run make test_e2e, which allows travis to finish running much faster

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
- [x] Administrative tasks are complete.
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: [LANTERN 222 JIRA Ticket](https://oncprojectracking.healthit.gov/support/browse/LANTERN-222)
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
- [x] Code review has been performed.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code accomplishes the tasks purpose.
- [x] Tests are complete.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.

### CODE REVIEW

**Comment by bmath10** [comment]
when I try running `make lint` I get the error: `there is no package called ‘lintr’`. Do I need to install the linter separately? If so, there should probably be instructions how to do that under the instructions for running the go linter.

File: scripts/lintr.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+cd ../shinydashboard/lantern
+echo "Running R lintr"
```
Comment (mayerm94) [inline scripts/lintr.sh:4]: I was able to install the lntr r package via this bash command. Try putting this into the script -- if it works then some of the new README text will be outdated
`echo 'install.packages("lintr", dependencies = TRUE, repos="http://cran.rstudio.com/")' | R --save`

Comment (emichaud998) [reply scripts/lintr.sh]: All set!
